lib/builder/builder.js

```diff
@@ -622,6 +623,22 @@ export default class Builder {
     this.customFilesWatcher = chokidar
       .watch(customPatterns, options)
       .on('change', refreshFiles)
+
+    // Watch for nuxt.config.js and user-defined changes
+    const restartServer = _.debounce((fname) => {
+      this.nuxt.callHook('watch:fileChanged', fname)
+    })
+
+    const nuxtRestartWatch = _.concat(
+      this.options.watch,
+      path.join(this.options.rootDir, 'nuxt.config.js')
+    )
+    this.nuxtRestartWatcher = chokidar
+      .watch(nuxtRestartWatch, options)
+      .on('change', (event, _path) => {
```

Comment:
@galvez `npm run dev` is crashing with the following error every time `nuxt.config.js` is modified:

```
path.js:39
    throw new ERR_INVALID_ARG_TYPE('path', 'string', path);
    ^

TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type object
    at assertPath (path.js:39:11)
    at Object.parse (path.js:910:5)
    at FSWatcher.watchers.restart.chokidar.watch.on (E:\Repositories\website\node_modules\nuxt-edge\dist\nuxt.js:3868:42)
    at FSWatcher.emit (events.js:182:13)
    at FSWatcher.EventEmitter.emit (domain.js:442:20)
    at FSWatcher.<anonymous> (E:\Repositories\website\node_modules\chokidar\index.js:199:15)
    at FSWatcher._emit (E:\Repositories\website\node_modules\chokidar\index.js:241:5)
    at FSWatcher.<anonymous> (E:\Repositories\website\node_modules\chokidar\lib\nodefs-handler.js:267:16)
    at FSReqWrap.oncomplete (fs.js:159:5)
```

According the [chokidar docs](https://github.com/paulmillr/chokidar#getting-started), the first argument should be the path and the second one the stats:

```javascript
// 'add', 'addDir' and 'change' events also receive stat() results as second
// argument when available: http://nodejs.org/api/fs.html#fs_class_fs_stats
watcher.on('change', (path, stats) => {
  if (stats) console.log(`File ${path} changed size to ${stats.size}`);
});
```

Reply:
@simonbrunel we just merged a fix for this, thanks again.

Reply:
Thanks :)